projects/js-packages/charts/src/components/line-chart/line-chart.tsx

```diff
@@ -246,7 +248,8 @@ const LineChart: FC< LineChartProps > = ( {
 } ) => {
 	const providerTheme = useChartTheme();
 	const theme = useXYChartTheme( data );
-	const chartId = useId(); // Ensure unique ids for gradient fill.
+	const internalChartId = useId(); // Ensure unique ids for gradient fill.
+	const chartId = useChartId( providedChartId );
```

Comment:
Do we need `internalChartId` anymore?

Reply:
> I kept them separate because `chartId` can be user-provided (for features like standalone legends: `<ChartLegend chartId="sales-chart" />`)

Ah I missed the fact that we support `providedId`. I think this is fine as is then.